tests/unittests/DeferredWeightLoaderTest.cpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) Glow Contributors. See CONTRIBUTORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Runtime/DeferredWeightLoader.h"
+#include "glow/ExecutionContext/ExecutionContext.h"
+#include "glow/ExecutionEngine/ExecutionEngine.h"
+
+#include "gtest/gtest.h"
+
+using namespace glow;
+using namespace glow::runtime;
+
+class TestDeferredWeightLoader : public DeferredWeightLoader {
+public:
+  Error loadNextWeight() override {
+    position_++;
+    return Error::success();
+  }
+  Error setSrc(void *loaderObject) override { return Error::success(); }
+  void addWeight(Tensor *weight) { weights_.push_back(weight); }
+  void addName(std::string name) { names_.push_back(name); }
+
+  std::string getName() override {
+    for (auto na : names_) {
+    }
+    if (position_ >= names_.size()) {
+      return "";
+    }
+    return names_[position_];
+  }
+
+  Tensor *getTensor() override {
+    if (position_ >= weights_.size()) {
+      return nullptr;
+    }
+    return weights_[position_];
+  }
+
+private:
+  std::vector<Tensor *> weights_{};
+  std::vector<std::string> names_{};
+  int position_{-1};
+};
+
+class DeferredWeightLoaderTest : public ::testing::Test {};
+
+std::unique_ptr<HostManager>
+createHostManager(llvm::StringRef backendName,
+                  HostConfig hostConfig = HostConfig()) {
+  std::vector<std::unique_ptr<DeviceConfig>> configs;
+  auto deviceConfig = glow::make_unique<DeviceConfig>(backendName);
+  configs.push_back(std::move(deviceConfig));
+  std::unique_ptr<HostManager> hostManager =
+      glow::make_unique<HostManager>(std::move(configs), hostConfig);
+  return hostManager;
+}
+
+TEST_F(DeferredWeightLoaderTest, staticPlaceholderInference) {
+
```

Comment:
Looks like this test is missing   CHECK_IF_ENABLED(); here.

Reply:
Good catch, it's currently hardcoded to OpenCL, I'll generalize it and add CHECK_IF_ENABLED().